src/common/config-parser-xml.c

```diff
@@ -571,10 +605,44 @@ static bool parse_audio_policy_configuration(struct parser_data *data,
     return true;
 }
 
+static bool parse_module_include(struct parser_data *data, const char *element_name, const XML_Char **attributes) {
+    struct includes *i;
+    char *href = NULL;
+    char *s, *t;
+    bool parsed = false;
+
+    if (!get_element_attr(data, attributes, true, ATTRIBUTE_href, &href)) {
+        pa_log("[%s:%u] Include but no href.", data->fn, data->lineno);
+        goto done;
+    }
+
+    /* We ignore xpointer attribute for now and just use the module element
+     * we are currently in when parsing the included file. */
+
+    i = pa_xmalloc0(sizeof(*i));
+    i->module = data->current_module;
+    i->href = href;
+    href = NULL;
+
+    SLLIST_APPEND(struct includes, data->conf->includes, i);
+    parsed = true;
+
+done:
+    pa_xfree(href);
```

Comment:
Is href ever something else than NULL here? In goto the href remains NULL because that is the fail path and above href is assigned NULL.

Reply:
You are right, and the implementation could be refactored otherwise a bit as well.. I implemented xpointer parsing first but decided to drop it in the end as it felt unnecessary complex for now and that left this function a bit odd.

Reply:
Fixed in updated PR.